backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/IteratorHandler.scala

```diff
@@ -170,21 +170,12 @@ class IteratorHandler extends IteratorApi with Logging {
         res
       }
 
-      override def next(): Any = {
-        if (!hasNext) {
-          throw new java.util.NoSuchElementException("End of stream")
```

Comment:
why remove it ?

Reply:
I think Iterator ensure call order, and each `resIter.hasNext` operation will lookup native static map that not cheap.

Reply:
I agree it would save native calls, but better to let author @zhztheplayer take a look if it's safe enough

Reply:
I'd suggest keeping the guard code like this unless we have special reason to remove it...

Since I don't make it sure that in gluten we always follow the legacy iterator convention, somehow a developer may write some code that calls next() without calling hasNext() at first.

For example you can look at the code in `ColumnarBatchInIterator.scala`, `next()` call will always return the same batch until `hasNext()` is called, that's definitely risky and not following the convention, but we don't have time to conduct a clean-up over the bad cases like that. So let's keep the guard code, temporarily?

Reply:
Make sense for me, thanks!